scripts/troubleshoot/LogCollection/AKSInsightsLogCollection.sh

```diff
@@ -0,0 +1,226 @@
+    fi
+
+    cmd=`kubectl exec ${rs_pod} -n kube-system -- ls /etc/opt/microsoft/docker-cimprov 2>&1`
+    if [[ $cmd == *"cannot access"* ]];then
+        echo -e "${Red}/etc/opt/microsoft/docker-cimprov not exist on ${rs_pod}${NC}" | tee -a Tool.log
+    else
+        kubectl cp ${rs_pod}:/etc/opt/microsoft/docker-cimprov/td-agent-bit-rs.conf omsagent-replicaset/td-agent-bit.conf --namespace=kube-system --container omsagent > /dev/null
+        kubectl cp ${rs_pod}:/etc/opt/microsoft/docker-cimprov/telegraf-rs.conf omsagent-replicaset/telegraf-rs.conf --namespace=kube-system --container omsagent > /dev/null
+    fi
+    echo -e "Complete log collection from ${rs_pod}!" | tee -a Tool.log
+}
+
+other_logCollection()
+{
+    echo -e "Collecting onboard logs..."
+    export deploy=$(kubectl get deployment --namespace=kube-system | grep -E omsagent | head -n 1 | awk '{print $1}')
+    if [ -z "$deploy" ];then
+        echo -e "${Red}there is not omsagent deployment, skipping log collection of deployment${NC}" | tee -a Tool.log
+    else
+        kubectl get deployment $deploy --namespace=kube-system -o yaml > deployment_${deploy}.txt
+    fi
+
+    export config=$(kubectl get configmaps --namespace=kube-system | grep -E container-azm-ms-agentconfig | head -n 1 | awk '{print $1}')
+    if [ -z "$config" ];then
+        echo -e "${Red}configMap named container-azm-ms-agentconfig is not found, if you created configMap for omsagent, please manually save your custom configMap of omsagent by command: kubectl get configmaps <configMap name> --namespace=kube-system -o yaml > configMap.yaml${NC}" | tee -a Tool.log
+    else
+        kubectl get configmaps $config --namespace=kube-system -o yaml > ${config}.yaml
+    fi
+
+    kubectl get nodes > node.txt
+    echo -e "Complete onboard log collection!" | tee -a Tool.log
+}
+
+#main
+output_path="AKSInsights-logs.$(date +%s).`hostname`"
+mkdir -p $output_path
+cd $output_path
+
+init
+
+export ds_pod=$(kubectl get pods -n kube-system -o custom-columns=NAME:.metadata.name | grep -E omsagent-[a-z0-9]{5} | head -n 1)
+if [ -z "$ds_pod" ];then
+	echo -e "${Red}daemonset pod do not exist, skipping log collection for daemonset pod${NC}" | tee -a Tool.log
+else
+    ds_logCollection
+fi
+
+export ds_win_pod=$(kubectl get pods -n kube-system -o custom-columns=NAME:.metadata.name | grep -E omsagent-win-[a-z0-9]{5} | head -n 1)
+if [ -z "$ds_win_pod" ];then
+	echo -e "${Cyan} windows agent pod do not exist, skipping log collection for windows agent pod ${NC}" | tee -a Tool.log
+else
+    win_logCollection
+fi
+
+export rs_pod=$(kubectl get pods -n kube-system -o custom-columns=NAME:.metadata.name | grep -E omsagent-rs-[a-z0-9]{5} | head -n 1)
+if [ -z "$rs_pod" ];then
+	echo -e "${Red}replicaset pod do not exist, skipping log collection for replicaset pod ${NC}" | tee -a Tool.log
+else
+    rs_logCollection
+fi
+
+other_logCollection
+
+cd ..
+echo
```

Comment:
did you validate this script works as expected? or do you want me to validate before approving?

Reply:
yes, I validated the script and it works well as expected. of course you can kindly double check and confirm